### PR TITLE
Autofocus event title on popover

### LINKED ISSUE #1930: new event focus directly on subject
when creating a new event, the focus should be on the subject so that the user can write the content without additional clicks.

**Describe the solution you'd like**
cursor on the yellow field (see attach) without other steps (additional click)

nc3

TY

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/87723053-new-event-focus-directly-on-subject?utm_campaign=plugin&utm_content=tracker%2F45525646&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F45525646&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #1930

### CODE REVIEW

File: src/directives/focus.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+export default {
+	inserted(el) {
+		setTimeout(() => {
```
Comment (georgehrke) [inline src/directives/focus.js]: It works with a  timeout, but not without.
There must be something, which takes away the focus immediately.

@tcitworld Do you have any idea what that could be?
My best guess would be that it's related to V-Popover, but that requires further debugging.

**Comment by laurens94** [comment]
Perhaps the autofocus behaviour should only take place whenever the user creates a new event (not when opening an existing event), to prevent mistakes. Or is that already the case with this commit?

**Comment by georgehrke** [comment]
@laurens94 Why would you not focus the title when editing an existing an event?

**Comment by laurens94** [comment]
Nevermind! I often edit an event just to change the time, but I guess it doesn't matter that much if the title is only focused instead of completely selected. 

Thanks for the fix 👍

**Comment by jancborchardt** [comment]
So is this ready to merge? :) @nextcloud/calendar

**Comment by Zaxooy** [comment]
Worked for me, can recommend this. :-)

I'm not very exerienced with github. So we're waiting for @tcitworld to merge this?

**Comment by ChristophWurst** [comment]
@GretaD could you rebase and give this a good test? I think we can integrate this then :)

**Review by GretaD — APPROVED** [review]

**Pull request merged**